plugins/dfu/fu-dfu-target.c

```diff
@@ -839,6 +839,8 @@ fu_dfu_target_upload_element_dfu(FuDfuTarget *self,
 		/* keep a sum of all the chunks */
 		chunk_size = (guint32)g_bytes_get_size(chunk_tmp);
 		total_size += chunk_size;
+		if (total_size > maximum_size)
+			break;
 
```

Comment:
this actually happen?  presumably we should probably log something as a warning or something, no?

Reply:
It happened when I broke the emulation code! :) We actually get a proper GError set further down.